src/g_game.c

```diff
@@ -1463,10 +1463,17 @@ static void G_DoPlayDemo(void)
 
   ExtractFileBase(defdemoname,basename);           // killough
 
-  lumpnum = W_GetNumForName(basename);
-  lumplength = W_LumpLength(lumpnum);
-
-  demobuffer = demo_p = W_CacheLumpNum(lumpnum, PU_STATIC);  // killough
+  lumpnum = W_CheckNumForName(basename);
+  if (lumpnum >= 0)
```

Comment:
My idea back then was to rename all demo lumps with a name collision to DEMO1:
https://github.com/fabiangreffrath/crispy-doom/blob/8b267f1d40166455eced8e19522a0b1756883833/src/doom/d_main.c#L1957

Reply:
Perhaps I'm describing the problem incorrectly. I recently ran some automated tests for `complevel vanilla` and there are several demos with names like demo_67801.lmp demo_67802.lmp etc. PrBoom+ loads these demos directly. I need to test Crispy as well.

Reply:
But can't these demos get loaded from the lumps if their names got truncated to e.g. `demo_678`? What my approach tries to solve is if you've got a demo called e.g. `map01.lmp`.

Reply:
> But can't these demos get loaded from the lumps if their names got truncated to e.g. `demo_678`? 

You are right, but for some reason port cannot find them. I will investigate more later.